src/settings/Token.js

```diff
@@ -7,26 +7,72 @@ import { stripesShape } from '@folio/stripes/core';
 import { ConfigForm } from '@folio/stripes/smart-components';
 import { Col, Row, TextField } from '@folio/stripes/components';
 
-
+/**
+ * Display decoded JWT token. Set a new token.
+ */
 class Token extends React.Component {
   static propTypes = {
     label: PropTypes.string.isRequired,
     stripes: stripesShape.isRequired,
   };
 
-  constructor() {
-    super();
-    this.onSave = this.onSave.bind(this);
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      raw: props.stripes.store.getState().okapi.token,
+      cooked: {},
+      error: '',
+    };
+  }
+
+  /**
+   * decode the original token
+   */
+  componentDidMount = () => {
```

Comment:
What goes wrong if we just do this right in the constructor?

Reply:
`parseToken` itself writes the decoded token (or an error) to state, and you can't call `setState` on a component until it's fully mounted, hence doing that work here in `cDM`. Surely there's a way I could refactor that a bit to return a value or throw an exception, or even better refactor the whole component to be functional, but this seemed the least amount of work possible, and given this is totally unsanctioned work**, that seemed best. 

**I don't even remember why but for some reason I was decoding tokens the other day, I think related to the refresh-token work in FOLIO-1233, and it annoyed me that I had to go to a third-party site to look at my own data.

Reply:
Yep, that all makes sense. Thanks again for doing this. It's great to see it now on https://folio-snapshot.aws.indexdata.com/settings/developer/token